FindAllShortestPaths.cpp

```diff
@@ -0,0 +1,198 @@
+    {
+        std::pair<int, int> s = stack.top();
+
+        onePath.push_back(s.first);
+
+        // if `s.first` is equal to toVertex then we need to save the answer and go back to see if there is more paths
+        if (s.first == toVertex)
+        {
+            std::vector<int> reversedOnePath(onePath.rbegin(), onePath.rend());
+            answer.push_back(reversedOnePath);
+        }
+
+        bool s_has_neighbors = false;
+
+        /*
+         we get from `Parents[s.first]` next index that we need. For example if `s` is pair<6, 2> then we need to push in stack Parents[6][2] and change
+         `s` to pair<6, 3>
+        */
+        if (s.second < Parents[s.first].size())
+        {
+            s_has_neighbors = true;
+            std::pair<int, int> helper = Parents[s.first][s.second];
+            helper.second = 0;
+            stack.push(helper);
+        }
+
+        if (!s_has_neighbors)
+        {
+            stack.pop();
+            if (stack.empty())
+            {
+                return;
+            }
+            s = stack.top();
+            ++s.second;
+            stack.pop();
+            stack.push(s);
+            onePath.resize(onePath.size() - 2);
+        }
+    }
+}
+
+std::vector<std::vector<int>> RecoverPaths(const std::vector<std::vector<std::pair<int, int>>>& Parents, int toVertex, int fromVertex)
+{
+    std::vector<std::vector<int>> answer;
+    std::vector<int> onePath;
+    RecoverPathsHelper(Parents, toVertex, fromVertex, answer, onePath);
+
+    return answer;
+}
+
+adjacency_list ReadGraph()
+{
+    int vertices, edges;
+    std::cin >> vertices >> edges;
+
+    adjacency_list Graph(vertices);
+
+    for (int i = 0; i < edges; ++i)
+    {
+        int a, b, weight;
+        std::cin >> a >> b >> weight;
+
+        Graph[a].push_back(std::make_pair(b, weight));
+        Graph[b].push_back(std::make_pair(a, weight));
+    }
+
+    return Graph;
+}
+
+void PrintGraph(const adjacency_list& Graph)
+{
+    int index = 0;
+    for (auto x : Graph)
+    {
+        std::cout << index << ": ";
+        for (auto y : x)
+        {
+            std::cout << y.first << "(" << y.second << ")  ";
+        }
+        std::cout << "\n";
+        ++index;
+    }
+}
+
+int main()
+{
+    adjacency_list Graph = ReadGraph();
+
+    int fromVertex, toVertex;
+
+    std::cin >> fromVertex >> toVertex;
+
+    std::vector<std::vector<std::pair<int, int>>> NotParsedPaths = ModifiedDijkstra(Graph, fromVertex);
+
+    std::vector<std::vector<int>> Paths = RecoverPaths(NotParsedPaths, fromVertex, toVertex);
+
+    /*
+    for (int i = 0; i < Paths.size(); ++i)
+    {
+        for (int j = 0; j < Paths[i].size(); ++j)
+        {
+            std::cout << Paths[i][j] << " ";
+        }
+        std::cout << "\n";
+    }
+    */
+
+
+    std::cout << "\n\n\nAll shortest paths:\n";
+    for(auto x : Paths)
+    {
+        for(auto y : x)
+        {
+            std::cout << y << " ";
+        }
+        std::cout << "\n";
```

Comment:
google what is `std::endl`.  it is a good thing

Reply:
done